src/dhcp/dnsmasq.c

```diff
@@ -75,16 +75,24 @@ int define_dhcp_interface_name(const struct dhcp_conf *dconf, int vlanid,
   }
 
 #ifdef WITH_UCI_SERVICE
-  snprintf(ifname, IFNAMSIZ, "%s%d", dconf->bridge_prefix, vlanid);
+  snprintf(ifname, IFNAMSIZ, "%.11s%d", dconf->bridge_prefix, vlanid);
```

Comment:
I'm not really sure how to do it safely.

Ideally we'd use something like `IFNAMSIZ - MAX_VLANID_CHARS - 1`, so that on platforms where `IFNAMSIZ` > 16, it auto expands, but I can't think of a good/easy/secure way of creating a `printf` specifier from a variable.

Reply:
How about `%.*`?

Reply:
`snprintf(ifname, IFNAMSIZ, "%.*s%d", IFNAMSIZ - MAX_VLANID_CHARS - 1, dconf->bridge_prefix, vlanid);`

Reply:
Implemented in https://github.com/nqminds/edgesec/pull/269/commits/a4698c9fa551b13536d8e820324541f620e72365!